From 6e82274dbe4aa950ac27e4573186bbf1f63ba367 Mon Sep 17 00:00:00 2001
From: Evan Severson <evanseverson@google.com>
Date: Tue, 30 Nov 2021 18:19:18 -0800
Subject: [PATCH] Hide overlays on ReviewPermissionsAtivity

Test: atest PermissionReviewTapjackingTest
Bug: 176094367
Merged-In: I9f263b947853e14d081a73ce907917e9326b6ef7
Change-Id: I9f263b947853e14d081a73ce907917e9326b6ef7
(cherry picked from commit cb6a249e1af72c89adf0f3f3179226723d8d389f)
Merged-In: I9f263b947853e14d081a73ce907917e9326b6ef7
---
 .../permission/ui/ReviewPermissionsActivity.java              | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/PermissionController/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java b/PermissionController/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
index dac012d1f..2054943f1 100644
--- a/PermissionController/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
+++ b/PermissionController/src/com/android/permissioncontroller/permission/ui/ReviewPermissionsActivity.java
@@ -21,6 +21,7 @@ import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.os.Bundle;
 import android.text.TextUtils;
+import android.view.WindowManager;
 
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentActivity;
@@ -41,6 +42,9 @@ public final class ReviewPermissionsActivity extends FragmentActivity
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        getWindow().addSystemFlags(
+                WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         PackageInfo packageInfo = getTargetPackageInfo();
         if (packageInfo == null) {
             finish();
-- 
2.38.0.413.g74048e4d9e-goog

